### PR TITLE
Added actions for cup stacking YCB task

### PR DESCRIPTION
**Pull request by vinitha910**
This branch contains stacking.py and actions.py. stacking.py contains the YCB cup stacking actions and actions.py contains general actions: moving objects and push grasping a cup.

### CODE REVIEW

**Comment by vinitha910** [comment]
actions.py should be removed from this branch. The feature/pitcher_pouring branch contains an updated version of actions.py

**Comment by mkoval** [comment]
See my comments on #78 - most of the same ones apply here.

**Comment by jeking04** [comment]
Okey dokey, @mkoval, I think we have this cleaned up. Can you take another look?

File: src/herbpy/action/stacking.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    move_cup_y, y_direc = MoveCupDistAndDirec(cup, stack, 1);
+    manip.PlanToEndEffectorOffset(direction=[0, y_direc, 0], distance = move_cup_y, position_tolerance=0.1, execute=True)
+
+    #Move cup down
+    with Disabled(stack):
+        manip.PlanToEndEffectorOffset(direction=[0, 0, -1], distance = 0.02, position_tolerance=0.1, execute=True)
+
+    #Release cup
+    manip.hand.OpenHand()
+    robot.Release(cup)
```
Comment (mkoval) [inline src/herbpy/action/stacking.py]: Missing lock.

File: src/herbpy/action/stacking.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+def StackCups(robot, table, cup, stack, cups_stacked):
+    """
+    @param robot The robot performing the stacking
+    @param cup The cup that is being stacked
+    @param stack The stack of cups the cup is going to be placed on
+    @param cups_stacked A list of the cups that have been stacked 
+    """
+	#Grasp cup
+    env = robot.GetEnv()
+    robot.PushGrasp(cup, yaw_range=[0, numpy.pi])
```
Comment (mkoval) [inline src/herbpy/action/stacking.py]: Why isn't `yaw_range` `[-pi, pi]`?

File: src/herbpy/action/stacking.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    robot.Release(cup)
+
+    with env:
+        cup.SetTransform(stack.GetTransform())
+    
+    #Add cup to list of cups stacked
+    cups_stacked.append(cup)
+    return cups_stacked
+
+def MoveCupDistAndDirec(cup, stack, direction):
```
Comment (mkoval) [inline src/herbpy/action/stacking.py]: Please rename to `MoveCupDistanceAndDirection`.

File: src/herbpy/action/stacking.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        manip = robot.GetActiveManipulator()
+
+    from prpy.rave import Disabled
+    with Disabled(table):
+        #Lift up cup
+        manip.PlanToEndEffectorOffset(direction=[0, 0, 1], distance = stack_height, position_tolerance=0.1, execute=True)
+
+    #Move cup in x-direction
+    move_cup_x, x_direc = MoveCupDistAndDirec(cup, stack, 0);
+    manip.PlanToEndEffectorOffset(direction=[x_direc, 0, 0], distance = move_cup_x, position_tolerance=0.1, execute=True)
```
Comment (mkoval) [inline src/herbpy/action/stacking.py]: I don't think it's safe to assume the cups can be moved purely in the `x` and `y` axes. Shouldn't the direction be the vector from the cup to the stack in the world frame?

**Comment by mkoval** [comment]
I added a few comments above. Also, Travis is failing with this error:
```
TypeError: fuze_grasp() got an unexpected keyword argument 'yaw_range'
```

**Comment by vinitha910** [comment]
@mkoval We made more changes and travis is happy, can you take another look? Thanks

File: src/herbpy/action/stacking.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    @param cups_stacked A list of the cups that have been stacked 
+    """
+	#Grasp cup
+    env = robot.GetEnv()
+
+    with env:
+        aabb_cup = cup.ComputeAABB()
+        aabb_stack = stack.ComputeAABB()
+        stack_height = 2*aabb_stack.extents()[2] + (len(cups_stacked)+1)*0.03
+        manip = robot.GetActiveManipulator()
```
Comment (mkoval) [inline src/herbpy/action/stacking.py]: Please take `manip` as an argument and only use `GetActiveManipulator` as the default argument.

File: src/herbpy/action/stacking.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        #Lift up cup
+        manip.PlanToEndEffectorOffset(direction=[0, 0, 1], distance = stack_height, position_tolerance=0.1, execute=True)
+
+    with env:
+        aabb_cup = cup.ComputeAABB()
+        aabb_stack = stack.ComputeAABB()
+    direc = numpy.array([0., 0., 0.])    
+    direc[:2] = aabb_stack.pos()[:2] - aabb_cup.pos()[:2]
+    dist = numpy.linalg.norm(direc)
+    print aabb_cup.pos(), aabb_stack.pos(), direc, dist
```
Comment (mkoval) [inline src/herbpy/action/stacking.py]: Please remove `print` statements. Replace them with `logging` statements if necessary.

**Comment by mkoval** [comment]
Looks great other than those two nitpicks. @vinitha910 thanks for making those changes!

**Comment by vinitha910** [comment]
@mkoval I made the changes, can you take another look? Thanks!

**Comment by mkoval** [comment]
Looks great! Thanks @vinitha910.

**Pull request merged**